code/obj/artifacts/artifact_objects/heatwave.dm

```diff
@@ -0,0 +1,34 @@
+
+/obj/artifact/heatwave
+	name = "artifact heatwave"
+	associated_datum = /datum/artifact/heatwave
+
+/datum/artifact/heatwave
+	associated_object = /obj/artifact/heatwave
+	type_name = "Heat Surge"
+	rarity_weight = 365
+	validtypes = list("ancient","martian","eldritch","precursor")
+	validtriggers = list(/datum/artifact_trigger/force,/datum/artifact_trigger/radiation,/datum/artifact_trigger/carbon_touch,/datum/artifact_trigger/silicon_touch,/datum/artifact_trigger/heat)
+	fault_blacklist = list(ITEM_ONLY_FAULTS,TOUCH_ONLY_FAULTS)
+	activ_text = "starts emitting HUGE flames!"
+	deact_text = "stops emitting flames."
+	react_xray = list(12,35,85,5,"POROUS") //has pores for flames idk
+	examine_hint = "It is covered in very conspicuous markings."
+	var/recharge_time = 20 SECONDS
+	var/recharging = FALSE
+	post_setup()
+		..()
+
+	effect_activate(var/obj/O)
+		if (..())
+			return
+		var/turf/T = get_turf(O)
+		if (recharging)
+			return
+		if (recharge_time > 0)
+			recharging = TRUE
+		playsound(O.loc, "sound/effects/mag_fireballlaunch.ogg", 50, 0)
+		T.visible_message("<span class='alert'><b>[O]</b> erupts into a huge column of flames! Holy shit!</span>")
+		fireflash_sm(T, 4, 7000, 2000)
```

Comment:
Oh, also, I feel like things are more interesting when there is more variance, so not every artifact of a type is the same.
So this could have some variance in how big or powerful the fireballs are, similar to attack_wand artifacts of the fire type.